[docs] Apply style guide to Data Grid Layout page (#9673)
mui/mui-x: docs/data/data-grid/layout/layout.md

<p class="description">The data grid offers multiple layout modes.</p>

:::error

> By default, the data grid has **no intrinsic dimensions**.

Reviewer: Always capitalize "Data Grid" whenever you see it, to reinforce the product name vs. the general concept of the thing. See: https://www.notion.so/mui-org/Style-conventions-for-MUI-components-and-products-dff7d2d7d1ba4a4abd47f48cf345b800

```suggestion
By default, the Data Grid has **no intrinsic dimensions**.
```

Author: Ahh thanks for the reminder! I made a similar change for "Date Picker"!

> Instead, it takes up the space given by its parent. The grid will raise an error in the console if its container has no intrinsic dimensions.

Reviewer: Add a line break to simplify line-by-line reviews. (It takes two line breaks for Markdown to start a new paragraph.)

```suggestion
Instead, it takes up the space given by its parent. 
The grid will raise an error in the console if its container has no intrinsic dimensions.
```

Author: Thanks for this! 

> It takes two line breaks for Markdown to start a new paragraph.

I see you only added one line break in the review if I'm not mistaken? Breaking to only to line 8 after ".. parent.". Your goal is to start a new paragraph?

Reviewer 2: No, it's a convention we use: 1 sentence = 1 line as you can see on the rest of the document (the line 6 you modified was an error).

The reason is to simplify the review process. If you have one sentence per line, reviewer can add one comment/suggestion per sentence which is easier to manage.

And it also produces more meaningful git diff because git does line by line comparison so with this convention it does a sentence-per-sentence comparison.

Author: @alexfauquette makes perfect sense. Thanks!

:::

## Percentage dimensions

> When using percentages (%) for height or width, make sure the container that you put your data grid in, has an intrinsic dimension. Browsers adjust the element based on a percentage of its parent's size. If the parent has no size, the percentage will be zero.

Reviewer: Formatting, copyediting, "data grid" => "Data Grid"

```suggestion
When using percentages (%) for height or width, make sure that the Data Grid's parent container has intrinsic dimensions. 
Browsers adjust the element based on a percentage of its parent's size. 
If the parent has no size, the percentage will be zero.
```

Author: Perfect! Thanks!

## Auto height

> The `autoHeight` prop enables the data grid to adjust its size based on its content.

Reviewer: ```suggestion
The `autoHeight` prop enables the Data Grid to adjust its size based on its content.
```

> This means that the height of the data grid will be determined by the number of rows, ensuring that all rows are visible to the user simultaneously.

Reviewer: Maintain future tense.

```suggestion
This means that the Data Grid's height will be determined by the number of rows, ensuring that all rows will be visible to the user simultaneously.
```
